wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -0,0 +1,654 @@
+         cipher == l_TLS_RSA_WITH_AES_128_CBC_SHA256 ||
+         cipher == l_TLS_RSA_WITH_AES_256_CBC_SHA ||
+         cipher == l_TLS_RSA_WITH_AES_256_CBC_SHA256) &&
+         side == WOLFSSL_CLIENT_END)
+        return 1;
+    else
+        return 0;
+}
+
+/* check if the g_alreadyVerified CA's key can be used for peer's certification */
+byte tsip_checkCA(word32 cmIdx)
+{
+    return (cmIdx == g_CAscm_Idx? 1:0);
+}
+
+/* check if tht root CA has been verified by TSIP, and it exists in the CM table.*/
+byte tsip_rootCAverified( )
+{
+    return (g_CAscm_Idx != (uint32_t)-1 ? 1:0);
+}
+
+/* open TSIP driver for use */
+int tsip_Open( ) {
+    
+    int ret;
+    
+    if ((ret = tsip_hw_lock()) == 0) {
+        /* open the TSIP */
+        ret = R_TSIP_Open((uint32_t*)s_flash, s_inst1, s_inst2);
+        if( ret != TSIP_SUCCESS ) {
+            WOLFSSL_MSG("RENESAS TSIP Open failed");
+        }
+        
+#if defined(WOLFSSL_RENESAS_TSIP_TLS)
+        /* generate TLS Rsa public key for Certificate verification */
+        if (ret == TSIP_SUCCESS && g_user_key_info.encrypted_user_tls_key) {
+            ret = R_TSIP_GenerateTlsRsaPublicKeyIndex(
+                    g_user_key_info.encrypted_session_key,
+                    g_user_key_info.iv,
+                    g_user_key_info.encrypted_user_tls_key,
+                    &g_user_key_info.user_rsa2048_tls_pubindex);
+            
+            if (ret != TSIP_SUCCESS) {
+                WOLFSSL_MSG("R_TSIP_GenerateTlsRsaPublicKeyIndex failed");
+            } else {
+                /* close once */
+                tsip_Close( );
+                /* open again with s_inst[] */
+                XMEMCPY(s_inst1, g_user_key_info.user_rsa2048_tls_pubindex.value, 
+                                                            sizeof(s_inst1));
+                ret = R_TSIP_Open((uint32_t*)s_flash, s_inst1, s_inst2);
+                if (ret != TSIP_SUCCESS) {
+                    WOLFSSL_MSG("R_TSIP_(Re)Open failed");
+                }
+                 /* init vars */
+                g_CAscm_Idx = (uint32_t)-1;
+            }
+        }
+#endif
+        /* unlock hw */
+        tsip_hw_unlock();
+    } else 
+        WOLFSSL_MSG("Failed to lock tsip hw \n");
+    
+    return ret;
+}
+
+/* close TSIP driver */
+void tsip_Close( ) {
+    int ret;
+    
+    if ((ret = tsip_hw_lock()) == 0) {
+        /* close TSIP */
+        ret = R_TSIP_Close();
+#if defined(WOLFSSL_RENESAS_TSIP_TLS)
+        g_CAscm_Idx = (uint32_t)-1;
+#endif
+        /* unlock hw */
+        tsip_hw_unlock();
+        if( ret != TSIP_SUCCESS ) {
+            WOLFSSL_MSG("RENESAS TSIP Close failed");
+        }
+    } else
+        WOLFSSL_MSG("Failed to unlock tsip hw \n");
+}
+
+/* Support functions for TSIP TLS Capability */
+#if defined(WOLFSSL_RENESAS_TSIP_TLS)
+
+/* to inform ca certificate sign */
+/* signature format expects RSA 2048 PSS with SHA256 */
+void tsip_inform_cert_sing(const byte *sign)
```

Comment:
Should this be named "tsip_inform_cert_sign" instead of "sing"?

Reply:
Yes, that's right. Thanks!